README.md

```diff
@@ -1,166 +1,151 @@
+- `ELASTICSEARCH_SERVICE_TOKEN`
+- `FLEET_SERVER_POLICY_ID`
 
-```
-SNAPSHOT=true DEV=true PLATFORMS=linux|darwin|windows mage package
-```
-
-The above assumes you are running on OS X. Set `PLATFORMS` to the one in you are running on. This speeds up packaging as it only builds it for your platform. As soon as this is completed (it might take a while for the first time) navigate to `build/distributions` and unpackage the `.tar.gz`. Change working directory to the elastic-agent directory and start the Elastic Agent:
-
-```
-KIBANA_HOST=http://localhost:5601 KIBANA_USERNAME=elastic KIBANA_PASSWORD=changeme ELASTICSEARCH_HOST=http://localhost:9200 ELASTICSEARCH_USERNAME=elastic ELASTICSEARCH_PASSWORD=changeme KIBANA_FLEET_SETUP=1 FLEET_SERVER_ENABLE=1 sudo ./elastic-agent container
+Create a self-signed TLS CA and cert+key for the fleet-server instance, you can use [elasticsearch-certutil](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html) for this:
+```shell
+# Create a CA
+../elasticsearch/bin/elasticsearch-certutil ca --pem --out stack.zip
+unzip stack.zip
+# Create a cert+key
+../elasticsearch/bin/elasticsearch-certutil cert --pem --ca-cert ca/ca.crt --ca-key ca/ca.key --ip $HOST_IP_ADDR --out cert.zip
+unzip cert.zip
 ```
 
-This will start up Elastic Agent with fleet-server and directly enroll it. In addition, Fleet is set up inside Kibana. In case the setup is done already in Kibana manually, the following command can be used:
+Ensure that `server.ssl.enabled: true` is set as well as the `server.ssl.certificate` and `server.ssl.key` attributes in `fleet-server.yml`
 
+Then run the fleet-server:
+```shell
+./build/binaries/fleet-server-8.7.0-darwin-x86_64/fleet-server -c fleet-server.yml
 ```
-sudo ./elastic-agent enroll --fleet-server=http://elastic:changeme@localhost:9200 --fleet-server-policy={fleet-server-policy-id} --enrollment-token={policy-enrollment-token}
-```
-
-## Running Elastic Agent with fleet-server in container
+By default the fleet-server will attempt to connect to Elasticsearch on `https://localhost:9200`, if this needs to be changed set it with `ELASTICSEARCH_HOSTS`
+The fleet-server should appear as an agent with the ID `dev-fleet-server`.
 
-If you want to run Elastic Agent and fleet-server in a container but built Kibana from source, you have to add the following to your `config/kibana.dev.yml`:
+Any additional agents will need the `ca/ca.crt` file to enroll (or will need to use the `--insecure` flag).
 
-```
-server.host: 0.0.0.0
-```
+#### fleet-server+agent on a Vagrant VM
 
-This makes sure, Kibana is accessible from the container. Start Kibana as before but for Elasticsearch, run the following command:
-
-```
-yarn es snapshot -E xpack.security.authc.api_key.enabled=true -E http.host=0.0.0.0
+The development Vagrant machine assumes the `elastic-agent`, `beats`, and `fleet-server` repos are in the same folder.
+Thus, it mounts `../` to `/vagrant` on the Vagrant machine.
```

Comment:
[Suggestion]
I'd keep the IP as it's hard coded in the Vagrant file, it won't change and is useful is someone is using the vagrant setup.

```suggestion
The development Vagrant machine assumes the `elastic-agent`, `beats`, and `fleet-server` repos are in the same folder.
Thus, it mounts `../` to `/vagrant` on the Vagrant machine. The vagrant machine IP address is `192.168.56.43`.
Use `https://192.168.56.43:8220` as fleet-server host.
```